src/negotiation/order.ts

```diff
@@ -73,18 +82,46 @@ export function assetOrderToSwap(orderAsset: OrderAsset): Asset | undefined {
   if (isNative(orderAsset)) {
     return {
       name: orderAsset.asset,
-      quantity: orderAsset.amount
+      quantity: orderAsset.nominalAmount
     };
   }
 
   const token = getToken(orderAsset.asset);
   if (token) {
     return {
       name: token.type.toLowerCase(),
-      quantity: orderAsset.amount,
+      quantity: orderAsset.nominalAmount,
       token_contract: token.address
     };
   }
 
   return undefined;
 }
+function areAmountEqual(
```

Comment:
seems this was branched from `negotation-follow-up` but this PR goes against `master`...

Reply:
Please see the PR's description.

Reply:
Touché 😛